go/vt/topo/vschema.go

```diff
@@ -53,6 +49,17 @@ func (ts *Server) SaveVSchema(ctx context.Context, keyspace string, vschema *vsc
 	return err
 }
 
+// ValidateAndSaveVSchema first validates the VSchema, then saves it.
+// If the VSchema is empty, just remove it.
+func (ts *Server) ValidateAndSaveVSchema(ctx context.Context, keyspace string, vschema *vschemapb.Keyspace) error {
+	_, err := vindexes.BuildKeyspace(vschema)
+	if err != nil {
```

Comment:
can not reuse the existing `SaveVSchema` and extend it to also validate the Vschema?

Reply:
I will revert `SaveVSchema` to what it was before. 

The idea behind splitting out validation from saving was to enable `ApplyVSchema` to validate when `--dry-run` is enabled without having `vindexes.BuildKeyspace` called multiple times.